File: nutritionalPanelFactExtraction/req_resources/TET/bind/java/extractor.java
/* PDF text extractor based on PDFlib TET
 *
 * $Id: extractor.java,v 1.31.2.1 2011/08/25 20:16:19 rjs Exp $
 */

import java.io.*;
import com.pdflib.TETException;
import com.pdflib.TET;
import java.text.DecimalFormat;

public class extractor
{
    /**
     * Global option list
     */
    static final String globaloptlist =
	    "searchpath={{../data} {../../../resource/cmap}}";
    
    /**
     * Document-specific option list
     */
    static final String docoptlist = "";
    
    /**
     * Page-specific option list
     */
    static final String pageoptlist = "granularity=page";
    
    /**
     * Separator to emit after each chunk of text. This depends on the
     * applications needs; for granularity=word a space character may be useful.
     */
    static final String separator = "\n";

    public static void main (String argv[])
    {
        TET tet = null;
        
	try
        {
	    if (argv.length != 2)
            {
                throw new Exception(
		    "usage: extractor <filename> <outfilename>");
            }

            Writer outfp = new BufferedWriter(new OutputStreamWriter(
                    new FileOutputStream(argv[1]), "UTF-8"));

            tet = new TET();

            tet.set_option(globaloptlist);

            int doc = tet.open_document(argv[0], docoptlist);

            if (doc == -1)
            {
                throw new Exception("Error " + tet.get_errnum() + "in "
                        + tet.get_apiname() + "(): " + tet.get_errmsg());
            }
            
            /* get number of pages in the document */
            int n_pages = (int) tet.pcos_get_number(doc, "length:pages");

            /* loop over pages in the document */
            for (int pageno = 1; pageno <= n_pages; ++pageno)
            {
                String text;
                int page;
		
		page = tet.open_page(doc, pageno, pageoptlist);

                if (page == -1)
                {
                    print_tet_error(tet, pageno);
                    continue; /* try next page */
                }

                /*
                 * Retrieve all text fragments; This is actually not required
                 * for granularity=page, but must be used for other
                 * granularities.
                 */
                while ((text = tet.get_text(page)) != null)
                {
                    /* print the retrieved text */
                    outfp.write(text);

                    /* print a separator between chunks of text */
                    outfp.write(separator);
                }


                if (tet.get_errnum() != 0)
                {
                    print_tet_error(tet, pageno);
                }

                tet.close_page(page);
            }

            tet.close_document(doc);
            outfp.close();
        }
	catch (TETException e)
	{
	    System.err.println("TET exception occurred in extractor sample:");
	    System.err.println("[" + e.get_errnum() + "] " + e.get_apiname() +
			    ": " + e.get_errmsg());
        }
        catch (Exception e)
        {
            System.err.println(e.getMessage());
        }
        finally
        {
            if (tet != null) {
		tet.delete();
            }
        }
    }

    /**
     * Report a TET error.
     * 
     * @param tet The TET object
     * @param pageno The page number on which the error occurred
     */
    private static void print_tet_error(TET tet, int pageno)
    {
        System.err.println("Error " + tet.get_errnum() + " in  "
                + tet.get_apiname() + "() on page " + pageno + ": "
                + tet.get_errmsg());
    }
}

File: nutritionalPanelFactExtraction/req_resources/TET/bind/.NET Framework 4.0/csharp/extractor.cs
// $Id: extractor.cs,v 1.24.2.1 2010/07/28 15:00:52 rjs Exp $
//
// PDF text extractor based on PDFlib TET
//

using System;
using System.IO;
using System.Text;
using TET_dotnet;

class Extractor {
  static int Main(string[] args) {

    /* global option list */
    string globaloptlist = "searchpath={{../data} {../../data}}";

    /* document-specific  option list */
    string docoptlist = "";

    /* page-specific option list */
    string pageoptlist = "granularity=page";

    /* separator to emit after each chunk of text. This depends on the
     * applications needs; for granularity=word a space character may be useful.
     */
    string separator = "\n";

    TET tet;
    FileStream outfile;
    BinaryWriter w;
    int pageno = 0;

    UnicodeEncoding unicode = new UnicodeEncoding(false, true);
    Byte[] byteOrderMark = unicode.GetPreamble();


    if (args.Length != 2)
    {
        Console.WriteLine("usage: extractor <infilename> <outfilename>");
        return(2);
    }

    outfile = File.Create(args.GetValue(1).ToString());
    w = new BinaryWriter(outfile);
    w.Write(byteOrderMark);

    tet = new TET();

    try
    {
        int n_pages;

        tet.set_option(globaloptlist);

        int doc = tet.open_document(args.GetValue(0).ToString(), docoptlist);

        if (doc == -1)
        {
            Console.WriteLine("Error {0} in {1}(): {2}",
                tet.get_errnum(), tet.get_apiname(), tet.get_errmsg());
            return(2);
        }

        /* get number of pages in the document */
        n_pages = (int) tet.pcos_get_number(doc, "length:pages");

        /* loop over pages in the document */
        for (pageno = 1; pageno <= n_pages; ++pageno)
        {
            string text;
            int page;

            page = tet.open_page(doc, pageno, pageoptlist);

            if (page == -1)
            {
                Console.WriteLine("Error {0} in {1}() on page {2}: {3}",
                    tet.get_errnum(), tet.get_apiname(), pageno,
                    tet.get_errmsg());
                continue;                        /* try next page */
            }

            /* Retrieve all text fragments; This is actually not required
             * for granularity=page, but must be used for other
             * granularities.
             */
            while ((text = tet.get_text(page)) != null)
            {
                /* print the retrieved text */
                w.Write(unicode.GetBytes(text));

                /* print a separator between chunks of text */
                w.Write(unicode.GetBytes(separator));
            }

            if (tet.get_errnum() != 0)
            {
                Console.WriteLine("Error {0} in {1}(): {3}",
                    tet.get_errnum(), tet.get_apiname(), tet.get_errmsg());
            }
            tet.close_page(page);
        }
        tet.close_document(doc);
    }
    catch (TETException e) {
        /* caught exception thrown by TET */
	Console.WriteLine("Error {0} in {1}(): {2}",
                e.get_errnum(), e.get_apiname(), e.get_errmsg());
    }
    catch (Exception e)
    {
        Console.WriteLine("General Exception: " + e.ToString());
        return(2);
    }
    finally
    {
        outfile.Close();
        if (tet != null) {
            tet.Dispose();
        }
    }

    return 0;
  }
}
